frontend/src/metabase/entities/collections.js

```diff
@@ -101,15 +101,14 @@ const Collections = createEntity({
     getInitialCollectionId: createSelector(
       [
         state => state.entities.collections,
+
         // these are listed in order of priority
-        (state, { collectionId }) => collectionId,
-        (state, { params }) => (params ? params.collectionId : undefined),
-        (state, { params, location }) =>
-          params && location && Urls.isCollectionPath(location.pathname)
-            ? Urls.extractCollectionId(params.slug)
-            : undefined,
-        (state, { location }) =>
-          location && location.query ? location.query.collectionId : undefined,
+        byCollectionIdProp,
+        byCollectionIdNavParam,
+        byCollectionUrlId,
+        byCollectionQueryParameter,
+
+        // defaults
         () => ROOT_COLLECTION.id,
         getUserPersonalCollectionId,
       ],
```

Comment:
You are making us dream here, @kulyk.
And this is good.

I can already see a one liner like

~~~js
getInitialCollectionId: createSelector(argument1, argument2),
~~~

yes, with better names for the arguments...but whoa this would be nice.

Reply:
Not sure it would make the selector more readable. It's handy to see an ordered list of id-lookup helpers and the selector body next to each other because the order of helpers is important here. If they are two separate things it won't be obvious they are connected so tightly

Reply:
The real goal would be to somehow improve function called in line 42

 `const Collections = createEntity({………`
 
 this is 125 lines long. 
 
 So yes, I see your point about this particular call to `createSelector`.
 
 Maybe we can improve in other ways…in the future.